tensorflow_addons/layers/multihead_attention_test.py

```diff
@@ -0,0 +1,166 @@
+
+        self.assertEqual(output.shape[1], q.shape[1])
+        self.assertEqual(output.shape[2], v.shape[2])
+
+    def test_no_value(self):
+        batch_size = 10
+        num_heads = 8
+        head_size = 12
+
+        q = tf.random.uniform((batch_size, 5, 9), dtype=np.float32)
+        k = tf.random.uniform((batch_size, 7, 11), dtype=np.float32)
+
+        mha = MultiHeadAttention(
+            head_size=head_size, num_heads=num_heads, return_attn_coef=True
+        )
+
+        output, attn_coef = mha([q, k])
+
+        self.assertEqual(attn_coef.shape[0], batch_size)
+        self.assertEqual(attn_coef.shape[1], num_heads)
+        self.assertEqual(attn_coef.shape[2], q.shape[1])
+        self.assertEqual(attn_coef.shape[3], k.shape[1])
+
+        self.assertEqual(output.shape[1], q.shape[1])
+        self.assertEqual(output.shape[2], k.shape[2])
+
+    def test_mask(self):
+        batch_size = 10
+        num_heads = 8
+        head_size = 12
+
+        q = tf.random.uniform((batch_size, 5, 9), dtype=np.float32)
+        k = tf.random.uniform((batch_size, 7, 11), dtype=np.float32)
+        v = tf.random.uniform((batch_size, 7, 13), dtype=np.float32)
+        mask = tf.random.uniform((batch_size, num_heads, 5, 7), dtype=np.float32) > 0.1
+
+        mha = MultiHeadAttention(
+            head_size=head_size, num_heads=num_heads, return_attn_coef=True
+        )
+
+        output, attn_coef = mha([q, k, v], mask=mask)
+
+        self.assertEqual(attn_coef.shape[0], batch_size)
+        self.assertEqual(attn_coef.shape[1], num_heads)
+        self.assertEqual(attn_coef.shape[2], q.shape[1])
+        self.assertEqual(attn_coef.shape[3], k.shape[1])
+
+        self.assertEqual(output.shape[0], batch_size)
+        self.assertEqual(output.shape[1], q.shape[1])
+        self.assertEqual(output.shape[2], v.shape[2])
+
+        if tf.executing_eagerly():
+            attn_coef = attn_coef.numpy()
+            mask = mask.numpy()
+
+            self.assertTrue(((attn_coef != 0) == mask).all())
+
+    def test_mask_no_batch(self):
+        batch_size = 10
+        num_heads = 8
+        head_size = 12
+
+        q = tf.random.uniform((batch_size, 5, 9), dtype=np.float32)
+        k = tf.random.uniform((batch_size, 7, 11), dtype=np.float32)
+        v = tf.random.uniform((batch_size, 7, 13), dtype=np.float32)
+        mask = tf.random.uniform((5, 7), dtype=np.float32) > 0.1
+
+        mha = MultiHeadAttention(
+            head_size=head_size, num_heads=num_heads, return_attn_coef=True
+        )
+
+        output, attn_coef = mha([q, k, v], mask=mask)
+
+        self.assertEqual(output.shape[0], batch_size)
+        self.assertEqual(output.shape[1], q.shape[1])
+        self.assertEqual(output.shape[2], v.shape[2])
+
+        if tf.executing_eagerly():
+            attn_coef = attn_coef.numpy()
+            mask = mask.numpy()
+
+            self.assertTrue(((attn_coef != 0) == mask).all())
+
```

Comment:
For layer behavior test, could u add:
1. small model with fit/eval/predict
2. from_config/to_config
3. save_model and load back.

Reply:
Added tests for all of these.